core/slice.go

```diff
@@ -978,3 +1028,196 @@ func (sl *Slice) AddPendingEtxsRollup(pEtxsRollup types.PendingEtxsRollup) error
 	}
 	return nil
 }
+
```

Comment:
Since this is a very specific set of functions can we move to a separate file? Maybe slice_fork.go?

Reply:
If we're going to split slice methods up across multiple files, then we need to group all of those files into a slice directory. Maybe something like this?
```
core/slice/append.go        // for the logic around appending blocks in a slice
core/slice/collections.go   // for all the ETX/manifest collection operations
core/slice/fork_recovery.go // for fork recovery logic in this PR
... other slice files?
```